src/lib.rs

```diff
@@ -30,62 +32,87 @@ pub struct Hx711<IN, OUT> {
     mode: Mode,
 }
 
-impl<IN, OUT> Hx711<IN, OUT>
+
+impl<IN, OUT, PINERR> Hx711<IN, OUT>
 where
-    IN: InputPin,
-    OUT: OutputPin,
+    IN: InputPin<Error = PINERR>,
+    OUT: OutputPin<Error = PINERR>,
 {
     /// Creates a new driver from Input and Outut pins
-    pub fn new(dout: IN, mut pd_sck: OUT) -> Self {
-        pd_sck.set_low();
-        let mut hx711 = Hx711 {
+    pub fn new(dout: IN, pd_sck: OUT) -> Self {
+        let hx711 = Hx711 {
             dout,
             pd_sck,
             mode: Mode::ChAGain128,
         };
-        hx711.reset();
         hx711
     }
 
-    /// Set the mode (channel and gain).
-    pub fn set_mode(&mut self, mode: Mode) {
-        self.mode = mode;
-        block!(self.retrieve()).unwrap();
+    /// Destruct the device and hand back the pins
+    pub fn destroy(self) -> (IN, OUT) {
+        (self.dout, self.pd_sck)
     }
 
-    /// Reset the chip. Mode is Channel A Gain 128 after reset.
-    pub fn reset(&mut self) {
-        self.pd_sck.set_high();
-        for _ in 1..3 {
-            self.dout.is_high();
-        }
-        self.pd_sck.set_low();
+    /// Enable 
+    pub fn enable(&mut self) -> Result<(), PINERR> {
+        self.pd_sck.set_low()
```

Comment:
Set mode after enable, because sleep mode resets the chip.

Reply:
True! Didn't think about that.